delira/utils/config.py

```diff
@@ -51,20 +547,219 @@ def nested_get(self, key, *args, **kwargs):
         else:
             return results[0]
 
-    def __setattr__(self, key, value):
+
+class DeliraConfig(LookupConfig):
+    """
+    Configure experiment for delira. Contains variables for model and training
+    which can be either fixed or variables (for hyperparameter search)
+    """
+
+    def __init__(self, dict_like=None, fixed_model=None, fixed_training=None,
+                 variable_model=None, variable_training=None, **kwargs):
         """
-        Modified to automatically convert `dict` to LookupConfig.
 
         Parameters
         ----------
-        key : str
-            the key which should be set
+        dict_like : dict, optional
+            dict like object containing values for config, by default None.
+        fixed_model : dict, optional
+            fixed parameters for model, by default None.
+        fixed_training : dict, optional
+            fixed parameters for training, by default None.
+        variable_model : dict, optional
+            variable parameters for model, by default None.
+        variable_training : dict, optional
+            variable parameters for training, by default None.
+        kwargs:
+            additional arguments added to the config
+        """
+        super().__init__(dict_like=dict_like, **kwargs)
+        self._update("fixed_model", self.generate_dict(fixed_model),
+                     overwrite=True)
+        self._update("fixed_training", self.generate_dict(fixed_training),
+                     overwrite=True)
+        self._update("variable_model", self.generate_dict(variable_model),
+                     overwrite=True)
+        self._update(
+            "variable_training",
+            self.generate_dict(variable_training),
+            overwrite=True)
+        self._version = get_versions()["version"]
+
+    @staticmethod
+    def generate_dict(value):
+        """
+        If value is none an emty dict will be created
+
+        Parameters
+        ----------
         value : Any
-            the corresponding value to set the key to.
+            checked value
+
+        Returns
+        -------
+        Any
+            dict if value is none otherwise value is returned
+        """
+        if value is None:
+            return {}
+        else:
+            return value
```

Comment:
maybe add an explicit conversion to `dict` here? This would make possible errors easier to track down since it's failure would be more clear.